Feature derive defmt for debug printing of enums

wallem89 opened the pull request:
To be able to print signals with an enum with `no_std` I added `defmt` to format the enum values so they can be printed for debugging right away. This is done with an optional flag to enable this feature.

projectgus commented:
@wallem89 I saw your PR and remembered I implemented `defmt` support a couple of months back, too. :facepalm: 

I've submitted it as #91. It's similar to yours but I also implemented `Format` for message structs.

wallem89 commented:
> @wallem89 I saw your PR and remembered I implemented `defmt` support a couple of months back, too. 🤦
> 
> I've submitted it as #91. It's similar to yours but I also implemented `Format` for message structs.

Your PR works for me too. And i like the `Format` for the message structs. Let's get yours merged and into a next release!

diondokter commented on Cargo.toml:
```diff
@@ -23,6 +23,7 @@ anyhow = "1.0.68"
 heck = "0.4.0"
 typed-builder = "0.18.0"
 embedded-can = "0.4.1"
+defmt = "0.3.8"
```
```suggestion
```

This doesn't need to be there. The dependency is not used in the code, and having it there prevents compilation on windows

wallem89 replied:
Should be fixed now, thanks!
